### PR TITLE
[incubator/alb-ingress-controller] Mirror chart from upstream

### PR DESCRIPTION
**Pull request by neerfri**
[alb-ingress-controller](https://github.com/kubernetes-sigs/aws-alb-ingress-controller) has a helm chart in the project's repo that is distributed from quay.io/coreos/alb-ingress-controller-helm using the [Helm Registry plugin](https://github.com/app-registry/helm-plugin).

Some users prefer to stick to the idiomatic way of installing charts without the need for an additional plugin.
https://github.com/kubernetes-sigs/aws-alb-ingress-controller/issues/362
https://github.com/kubernetes-sigs/aws-alb-ingress-controller/issues/282

This PR adds a mirror of the [upstream chart folder](https://github.com/kubernetes-sigs/aws-alb-ingress-controller/tree/master/alb-ingress-controller-helm) with an easy way to update from upstream (run `make update-from-upstream`)

I've replaced the README file to prevent misleading installation instructions from the upstream README. The upstream `README.md` is renamed to `README-upstream.md` to keep the information regarding the values of the chart in sync with the code in this repo.

### CODE REVIEW

**Comment by Jeeppler** [comment]
Is there any problem in merging this chart into the helm/incubator?

File: incubator/alb-ingress-controller/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
+appVersion: 1.0-beta.6
```
Comment (M00nF1sh) [inline incubator/alb-ingress-controller/Chart.yaml]: this should be changed to 1.0-beta.7(which contains multiple bug fixes) 😄

File: incubator/alb-ingress-controller/Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+update-from-upstream:
+	mv README.md README-tmp.md
```
Comment (M00nF1sh) [inline incubator/alb-ingress-controller/Makefile:2]: i don't think we need this. We should deprecate the helm repo in our alb-ingress-controller repo, and maintain helm templates in this helm/charts repo.

File: incubator/alb-ingress-controller/README-upstream.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# alb-ingress-controller
+
```
Comment (M00nF1sh) [inline incubator/alb-ingress-controller/README-upstream.md:2]: we can just remove this file

File: incubator/alb-ingress-controller/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# alb-ingress-controller
+
+alb-ingress-controller satisfies Kubernetes ingress resources by provisioning Application Load Balancers.
+
+This is a clone of the upstream helm chart located at:
+https://github.com/kubernetes-sigs/aws-alb-ingress-controller/tree/master/alb-ingress-controller-helm
```
Comment (M00nF1sh) [inline incubator/alb-ingress-controller/README.md:6]: we can delete this section(upstream clone), and always maintain the helm template here

File: incubator/alb-ingress-controller/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+## Installing the Chart
+
+```console
+$ helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com/
+$ helm install incubator/alb-ingress-controller
+```
+
+
+## Configuration
+
```
Comment (M00nF1sh) [inline incubator/alb-ingress-controller/README.md:18]: should move content from upstream to here

**Comment by neerfri** [comment]
Updated to `1.0-beta.7` as suggested by @M00nF1sh 
Also all commits are now signed off.
I believe any further changes to the documentation can and should be done in a next PR after this chart exists for users to get it.

**Comment by M00nF1sh** [comment]
> @M00nF1sh can you please review the chart again? Are there any outstanding issues you want to have addressed?
> 
> As @neerfri mentioned users are waiting for this PR to go in. This would allow users to easily deploy changes to EKS using the official Helm repository. In addition, it would users allow to use the Terraform [Helm provider](https://www.terraform.io/docs/providers/helm/index.html).

The chart looks good to me. (I don't have permissions to approve it), i tried to find helm members to review this in #charts slack, but haven't get luck so far : (

**Comment by Jeeppler** [comment]
@M00nF1sh can you become a member of Helm chart repository? As far as I can see you are working for Amazon and I think it would be good to have somebody from Amazon being able to maintain and approve changes for this chart even in the future. At the end it is an AWS specific chart and will most likely be used with either EKS or Kops.

What does @d-nishi thinks about having an active Amazon developer involved in Helm charts? She seems to be a manager at Amazon responsible for Open Source Software Development.

Have you tried to contact @mgoodness? He is both a top contributor to the ALB ingress controller project and Helm chart contributor.

**Comment by d-nishi** [comment]
@Jeeppler -- I think @M00nF1sh should be a Helm contributor and it is indeed a great idea.

**Comment by Jeeppler** [comment]
@d-nishi and @M00nF1sh any updates regarding becoming a contributor?

I really would like to see this pull request merged.

**Comment by Jeeppler** [comment]
@mgoodness and @prydonius do you think this chart can be merged?

**Comment by M00nF1sh** [comment]
> @d-nishi and @M00nF1sh any updates regarding becoming a contributor?
> 
> I really would like to see this pull request merged.

I haven't started the work to become members yet, guess will requires contributions. Will do it once features set of ALB ingress controller is stable.

**Comment by Jeeppler** [comment]
@M00nF1sh 

> Will do it once features set of ALB ingress controller is stable.

What feature set of ALB ingress controller has to be stable? What means stable from your perspective?

Keep in mind this pull request, tries to get the alb-ingress-controller into the incubator Helm repository. Not into stable. I personally, want to be able to use the alb-ingress-controller in my testing environment and report issues back, rather than waiting for a super stable version.

**Comment by Jeeppler** [comment]
@mgoodness and @prydonius is this pull request ready to get merged from your perspective?

**Comment by Jeeppler** [comment]
@unguiculus codecentric AG works with AWS. Are you able to help us out here by approving the pull request? Your help would be appreciated.

@scottrigby can you help us out to merge this pull request in?

@rimusz can you help us out to merge this pull request in?

**Comment by d-nishi** [comment]
@Jeeppler @M00nF1sh on this.

**Comment by d-nishi** [comment]
@mattfarina - as discussed can you please merge out helm chart on aws-alb-ingress-controller? Thanks a lot!

**Comment by Jeeppler** [comment]
1. Can you merge kubernetes-sigs/aws-alb-ingress-controller#772
2. Ask @neerfri to update this chart or if not possible create a new pull request

**Comment by M00nF1sh** [comment]
> @Jeeppler @M00nF1sh -- this PR is old and needs a refresh. It would be best to close this and open a new PR. @M00nF1sh - thoughts?

Yeah, i'll create a new PR on behalf for @neerfri, which is quicker, and @neerfri still take the credit for his work to help this out 😄

**Comment by d-nishi** [comment]
@mattfarina - this PR is old. Can you please close it?

The new PR is here #9813 -- assigned you to review it. Can you merge post review?

**Pull request closed without merging**